modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -323,12 +323,27 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
 
         return new_frame
 
-    def agg(self, agg):
+    def agg(self, agg, **kwargs):
+        """
+        Perform specified aggregation alogn columns.
+
+        Parameters
+        ----------
+        agg : str
+            Name of the aggregation function to perform.
+        **kwargs : dict
+            Additional parameters to pass to the aggregation expression.
+
+        Returns
+        -------
+        OmnisciOnRayFrame
+            New frame containing the result of aggregation.
+        """
         assert isinstance(agg, str)
 
         agg_exprs = OrderedDict()
         for col in self.columns:
-            agg_exprs[col] = AggregateExpr(agg, self.ref(col))
+            agg_exprs[col] = AggregateExpr(agg, self.ref(col), **kwargs)
```

Comment:
[`AggregateExpr`](https://github.com/modin-project/modin/blob/f9c99c3817d35f7bf0d91df9aa9644545ea27517/modin/experimental/engines/omnisci_on_ray/frame/expr.py#L263) has a `distinct` parameter, which indicates whether the `count` should be distinct. We're [forward](https://github.com/modin-project/modin/pull/3158/files#diff-ec1bdeb1039dcfbc857a3d672d2844ddb9d5deb32f6413584bb2d476c481ad8fR321) this parameter from the query compiler level.

Reply:
Do you intentionally use this construction to get an error in case some new parameter is added by the front-end to force explicit handling of all `kwargs` params? I just feel like extracting `distinct` arg and passing it directly would make the code simpler to understand.

Reply:
Yes, this was done on purpose so the error can indicate if some parameters are not processed at the level where they should be.

Explicit extraction of the `distinct` parameter looks not so good to me, because we're adding this corner case for a single operation in the general `agg` function.